src/interface_adapter/my_folder/MyFolderPresenter.java

```diff
@@ -0,0 +1,47 @@
+package interface_adapter.my_folder;
+
+import interface_adapter.ViewManagerModel;
+import interface_adapter.open_folder.OpenFolderState;
+import interface_adapter.open_folder.OpenFolderViewModel;
+import interface_adapter.opened_folder.OpenedFolderState;
+import interface_adapter.opened_folder.OpenedFolderViewModel;
+import use_case.my_folder.MyFolderOutputBoundary;
+import use_case.my_folder.MyFolderOutputData;
+import use_case.open_folder.OpenFolderOutputBoundary;
+import use_case.open_folder.OpenFolderOutputData;
+
+public class MyFolderPresenter implements MyFolderOutputBoundary {
+
+    private final MyFolderViewModel myFolderViewModel;
+    private final OpenedFolderViewModel openedFolderViewModel;
+    private ViewManagerModel viewManagerModel;
+
+    public MyFolderPresenter(ViewManagerModel viewManagerModel,
+                                   MyFolderViewModel myFolderViewModel,
+                                   OpenedFolderViewModel openedFolderViewModel) {
+        this.viewManagerModel = viewManagerModel;
+        this.myFolderViewModel = myFolderViewModel;
+        this.openedFolderViewModel = openedFolderViewModel;
+        }
+
+        @Override
+        public void prepareSuccessView(MyFolderOutputData response) {
+            // On success, switch to the logged in view.
+
+            OpenedFolderState openedFolderState = openedFolderViewModel.getState();
+            openedFolderState.setFoldername(response.getFoldername());
+            this.openedFolderViewModel.setState(openedFolderState);
+            this.openedFolderViewModel.firePropertyChanged();
```

Comment:
why did you use openedFolderView? shouldn't it be myFolderView for this use case?

Reply:
this means we transfer it to the opened folder view which is the next view of my folder view